File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        pos = []
        neg = []
        res = []
        for i in range(n):
            if(nums[i]>0):
                pos.append(nums[i])
            else:
                neg.append(nums[i])
        for i in range(len(pos)):
            res.append(pos[i])
            res.append(neg[i])
        return res

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int n = nums.length;
        ArrayList<Integer> post = new ArrayList<>();
        ArrayList<Integer> neg = new ArrayList<>();
        ArrayList<Integer> result = new ArrayList<>();
        int[] arr = new int[n];
        for(int i = 0 ; i< n ; i++){
            if(nums[i]>=0){
                post.add(nums[i]);
            }else{
                neg.add(nums[i]);
            }
        }
        for(int i = 0 ; i<post.size() ; i++){
            result.add(post.get(i));
            result.add(neg.get(i));

        }
        for(int i = 0 ; i<result.size();i++){
            arr[i] = result.get(i);
        }

        return arr;
        
    }
}